pymofa/experiment_handling.py

```diff
@@ -175,63 +338,83 @@ def compute(self, run_func, skipbadruns=False):
-                tag = self.status.Get_tag()
-                if tag == tags.READY:
-                    # node is ready, can take new task
-                    if task_index < len(self.tasks):
-                        self.comm.send(self.tasks[task_index], dest=source,
-                                       tag=tags.START)
-                        task_index += 1
-                    else:
-                        self.comm.send(None, dest=source, tag=tags.EXIT)
-                elif tag == tags.FAILED:
-                    # node failed to complete.
-                    # retry (failed runs send their task as return)
-                    self.comm.send(n_return, dest=source, tag=tags.START)
-                elif tag == tags.DONE:
-                    # node succeeded
                     tasks_completed += 1
+                    self._obtain_store_function(task)(result)
                     self._progress_report(tasks_completed, len(self.tasks),
                                           "Calculating...")
-                elif tag == tags.EXIT:
-                    # node completed all tasks. close
-                    closed_nodes += 1
-            print("Calculating 0 ...done.")
+            # = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
+            else:
+            # = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = = =
+                print("Splitting calculations to {} nodes."\
+                    .format(self.n_nodes))
+                sys.stdout.flush()
+
+                task_index = 0
+                closed_nodes = 0
+                while closed_nodes < self.n_nodes:
+                    n_return = self.comm.recv(source=MPI.ANY_SOURCE,
+                                              tag=MPI.ANY_TAG,
+                                              status=self.status)
+                    source = self.status.Get_source()
+                    tag = self.status.Get_tag()
+
+                    if tag == tags.READY:
+                        # node is ready, can take new task
+                        if task_index < len(self.tasks):
+                            self.comm.send(self.tasks.iloc[task_index], dest=source,
+                                           tag=tags.START)
+                            task_index += 1
+                        else:
+                            self.comm.send(None, dest=source, tag=tags.EXIT)
+
+                    elif tag == tags.FAILED:  # node failed to complete.
+                        # retry (failed runs send their task as return)
+                        self.comm.send(n_return, dest=source, tag=tags.START)
+
+                    elif tag == tags.DONE:
+                        # node succeeded
+                        tasks_completed += 1
+
+                        # store results
+                        # completed runs send thier (task, result) as return
+                        self._obtain_store_function(n_return[0])(n_return[1])
```

Comment:
I suggest moving this to slave nodes, since sending large packages of data between nodes might lead to unpickling errors (in my experience). Will open issue and commit solution.